app/components/ads/search-incontent-native.js

```diff
@@ -0,0 +1,39 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import Ads from '../../modules/ads';
+import AdsMixin from '../../mixins/ads';
+
+export default Component.extend(AdsMixin, {
+  ads: service(),
+
+  init() {
+    this._super(...arguments);
+
+    this.adsContextPromise = this.ads.fetchSearchAdsContext();
```

Comment:
Does it mean that we'll request search context twice? Once for top leaderboard and once for incontent_native?

Reply:
Probably we should synchronise this call between those two slots